compute_tools/src/compute.rs

```diff
@@ -695,4 +735,115 @@ LIMIT 100",
             "{{\"pg_stat_statements\": []}}".to_string()
         }
     }
+
+    // If remote extension storage is configured,
+    // download extension control files
+    pub async fn prepare_external_extensions(&self, compute_state: &ComputeState) -> Result<()> {
+        if let Some(ref ext_remote_storage) = self.ext_remote_storage {
+            let pspec = compute_state.pspec.as_ref().expect("spec must be set");
+            let spec = &pspec.spec;
+            let custom_ext = spec.custom_extensions.clone().unwrap_or(Vec::new());
+            info!("custom extensions: {:?}", &custom_ext);
+            let ext_remote_paths = extension_server::get_available_extensions(
+                ext_remote_storage,
+                &self.pgbin,
+                &self.pgversion,
+                &custom_ext,
+                &self.build_tag,
+            )
+            .await?;
+            self.ext_remote_paths
+                .set(ext_remote_paths)
+                .expect("ext_remote_paths.set error");
+        }
+        Ok(())
+    }
+
+    pub async fn download_extension(&self, ext_name: &str) -> Result<()> {
+        match &self.ext_remote_storage {
+            None => anyhow::bail!("No remote extension storage"),
+            Some(remote_storage) => {
+                // TODO: eliminate useless LOAD Library calls to this function (if possible)
+                // not clear that we can distinguish between useful and useless
+                // library calls better than the below code
```

Comment:
What do you mean by `useless LOAD library calls` ?

Reply:
> which may implicitly load library, that is a part of relation.
When this implicit library load happens, it calls `download_extension` for an extension that has already been downloaded
Like, I have `api.rs` set up so that it will listen for `LOAD library` calls and handle them by calling `download_extension`
I have a list of extensions that have already been downloaded, so this is not a huge deal, but it seems unnecessary to call `download_extension` again, even if we can exit quickly because the extension has already been downloaded. 

> there is also LOAD libname
yes, this is still supported

Reply:
I've added a test for LOAD 'anon'  and it seems to work fine.


>When this implicit library load happens, it calls download_extension for an extension that has already been downloaded

Postgres always tries to find library file locally first, and only if it is not found, requests it from `compute_ctl`.
So this shouldn't be the case.

In any case, I think this is fine to leave check inside `download_extension`.